packages/design-system-react-native/src/components/Checkbox/Checkbox.test.tsx

```diff
@@ -0,0 +1,227 @@
+/* eslint-disable @typescript-eslint/no-require-imports */
+import { useTailwind } from '@metamask/design-system-twrnc-preset';
+import { renderHook } from '@testing-library/react-hooks';
+import { render, fireEvent } from '@testing-library/react-native';
+import type { ReactElement } from 'react';
+import React, { createRef } from 'react';
+import { Text } from 'react-native';
+import type {
+  StyleProp,
+  ViewStyle,
+  PressableProps,
+  PressableStateCallbackType,
+} from 'react-native';
+
+import { Checkbox } from './Checkbox';
+
+describe('Checkbox', () => {
+  let tw: ReturnType<typeof useTailwind>;
+
+  beforeAll(() => {
+    tw = renderHook(() => useTailwind()).result.current;
+  });
+
+  beforeEach(() => {
+    jest.useFakeTimers();
+  });
+
+  afterEach(() => {
+    jest.runOnlyPendingTimers();
+    jest.useRealTimers();
+  });
+
+  /**
+   * Flatten a style prop into an array of style objects.
+   *
+   * @param styleProp - The style prop to flatten.
+   * @returns A list of style objects.
+   */
+  function flattenStyles(
+    styleProp: StyleProp<ViewStyle> | undefined,
+  ): ViewStyle[] {
+    if (styleProp === null || styleProp === undefined) {
+      return [];
+    }
+    if (Array.isArray(styleProp)) {
+      return styleProp.flatMap((item) =>
+        flattenStyles(item as StyleProp<ViewStyle>),
+      );
+    }
+    if (typeof styleProp === 'object') {
+      return [styleProp as ViewStyle];
+    }
+    return [];
+  }
```

Comment:
Created an issue to turn this into a testing utility https://github.com/MetaMask/metamask-design-system/issues/691